### PR TITLE
feat: handle automatic publication after validation

### PR DESCRIPTION
**Pull request by ReidyT**
Update the frontend to be compatible with (graasp/graasp#1077).

- Ask the user to set item to public during the validation (but will be public only if succeeded)
- refactor the PublicationButton into smaller components.

### CODE REVIEW

File: src/components/item/publish/publicationButtons/NotPublicButton.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    </LoadingButton>
+  );
+
+  return (
+    <>
+      <PublicVisibilityModal
+        item={item}
+        isOpen={isOpen}
+        onClose={closeModal}
+        onValidate={() => {}}
```
Comment (spaenleh) [inline src/components/item/publish/publicationButtons/NotPublicButton.tsx:39]: Mhmmm ?

File: src/components/item/publish/publicationButtons/ReadyToPublishButton.tsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    }
+  };
+
+  const handleModalValidate = () => {
+    publishItem();
+    closeModal();
+  };
+
+  const description = (
+    <Alert icon={<CheckIcon fontSize="inherit" />} severity="info">
```
Comment (spaenleh) [inline src/components/item/publish/publicationButtons/ReadyToPublishButton.tsx:54]: It bugs me that you use the `info` severity with the check ... why not use the `success` ?

File: src/components/item/publish/publicationButtons/PublicationButton.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    return <Typography>{description}</Typography>;
+  }
+
+  return description;
+};
+
+type Props = {
+  isLoading: boolean;
+  description: string | JSX.Element;
+  elements: JSX.Element | JSX.Element[];
```
[the remark this answers is not in the document]
Comment (ReidyT) [reply src/components/item/publish/publicationButtons/PublicationButton.tsx:18]: Thanks for this, I really didn't know, very interesting !

File: src/components/item/publish/publicationButtons/PublicationButton.tsx [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  isLoading,
+  description,
+  elements,
+}: Props): JSX.Element => (
+  <ContentLoader isLoading={isLoading}>
+    <Stack spacing={4}>
+      {getDescriptionElement(description)}
+
+      <Stack justifyContent="center" direction="row" spacing={2}>
+        {elements}
```
[the remark this answers is not in the document]
Comment (ReidyT) [reply src/components/item/publish/publicationButtons/PublicationButton.tsx:31]: Great idea !

File: src/components/item/publish/publicationButtons/NotPublicButton.tsx (hunk 1 above)
Comment (ReidyT) [reply src/components/item/publish/publicationButtons/NotPublicButton.tsx:39]: yeah I agree 😅

**Pull request merged**